### PR TITLE
Fix #7767: Added new PowerBi APIs and lineage

### LINKED ISSUE #7767: Revisit PowerBI lineage
so far, we have created lineage against the database. Take another look at the PBI API and document our findings. We need to end up with a doc explaining what we looked at and what we found. Let's also make sure that we create different datasets, from SQL or DAX and see if we can extract further information in these cases.

### PR DESCRIPTION
**Pull request by OnkarVO7**
### Describe your changes :

Fix https://github.com/open-metadata/OpenMetadata/issues/7767

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Frontend Preview (Screenshots) :
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@open-metadata/ingestion

### CODE REVIEW

**Review by pmbrull — COMMENTED** [review]
Thanks! left a couple of comments if you have the time

File: ingestion/src/metadata/ingestion/source/dashboard/powerbi.py [hunk 1]
```diff
@@ -151,31 +248,54 @@ def yield_dashboard_chart(
...
+
+    def fetch_dataset_from_workspace(self, dataset_id: str) -> Optional[dict]:
+        """
+        Method to search the dataset using id in the workspace dict
+        """
+
+        dataset_data = next(
+            (
+                dataset
+                for dataset in self.context.workspace.get("datasets")
```
[the remark this answers is not in the document]
Comment (OnkarVO7) [reply ingestion/src/metadata/ingestion/source/dashboard/powerbi.py:296]: Got it I'll validate this

File: ingestion/src/metadata/clients/powerbi_client.py [hunk 2]
```diff
@@ -76,67 +77,85 @@ def get_auth_token(self) -> Tuple[str, str]:
...
+        """
+        try:
+            data = json.dumps({"workspaces": workspace_ids})
+            path = (
+                "/myorg/admin/workspaces/getInfo?"
+                "datasetExpressions=True&datasetSchema=True"
+                "&datasourceDetails=True&getArtifactUsers=True&lineage=True"
+            )
+            response = self.client.post(path=path, data=data)
+            return response
```
[the remark this answers is not in the document]
Comment (OnkarVO7) [reply ingestion/src/metadata/clients/powerbi_client.py:124]: Yes

File: ingestion/src/metadata/ingestion/source/dashboard/powerbi.py [hunk 3]
```diff
@@ -45,6 +46,72 @@ class PowerbiSource(DashboardServiceSource):
...
+                    scan_id=workspace_scan_id
+                )
+                self.workspace_data = response.get("workspaces")
+            else:
+                logger.error("Error in fetching dashboards and charts")
+        else:
+            logger.error("Unable to fetch any Powerbi workspaces")
+        return super().prepare()
+
+    def wait_for_scan_complete(self, scan_id, timeout=100) -> bool:
```
[the remark this answers is not in the document]
Comment (OnkarVO7) [reply ingestion/src/metadata/ingestion/source/dashboard/powerbi.py:88]: I was thinking in PBI client we keep only the API calls. 
I can move it to the client file if required. Let me know @pmbrull

[the remark this answers is not in the document]
Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/dashboard/powerbi.py:88]: awesome thank you!

File: ingestion/src/metadata/ingestion/source/dashboard/powerbi.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (OnkarVO7) [reply ingestion/src/metadata/ingestion/source/dashboard/powerbi.py]: Added the logic here

**Review by pmbrull — APPROVED** [review]

**Pull request merged**